packages/mobile/src/language/Language.tsx

```diff
@@ -12,25 +12,31 @@ import { CustomEventNames } from 'src/analytics/constants'
 import { setLanguage } from 'src/app/actions'
 import { AVAILABLE_LANGUAGES } from 'src/config'
 import { Namespaces } from 'src/i18n'
+import { emptyHeader, headerWithBackButton } from 'src/navigator/Headers.v2'
 import { navigate, navigateBack } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 
-type Props = StackScreenProps<StackParamList, Screens.Language>
+type RouterProps = StackScreenProps<StackParamList, Screens.Language>
+type Props = RouterProps
 
 export default function Language({ route }: Props) {
   const dispatch = useDispatch()
   const { t, i18n } = useTranslation(Namespaces.accountScreen10)
+  const fromSettings = route.params?.fromSettings
 
   function onSelect(language: string, code: string) {
     CeloAnalytics.track(CustomEventNames.language_select, { language, selectedAnswer: code })
-    const nextScreen = route.params?.nextScreen ?? Screens.JoinCelo
     dispatch(setLanguage(code))
-    if (nextScreen === 'GO_BACK') {
-      navigateBack()
-    } else {
-      navigate(nextScreen)
-    }
+    // Wait for next frame before navigating back
+    // so the user can see the changed selection briefly
+    requestAnimationFrame(() => {
+      if (fromSettings) {
```

Comment:
i think the nextScreen pattern is more generic and gives more info about the behavior on selection - is there any reason to switch it?

Reply:
Adding a special case `GO_BACK` to Screens did not feel right for me, so I've tried to replace it with `if nextScreeen is defined - use it, otherwise navigateBack`. It is rather a proposal, so if folks would like it - I'm happy to revert.

Reply:
I'm fine either way personally :)